src/map/battle.cpp

```diff
@@ -2827,6 +2909,24 @@ static bool is_attack_critical(struct Damage* wd, struct block_list *src, struct
 			case WH_GALESTORM:
 				if (sc && !sc->data[SC_CALAMITYGALE])
 					return false;
+				break;
+			case SKE_NOON_BLAST:
+				if (!(sc && (sc->data[SC_NOON_SUN] || sc->data[SC_SKY_ENCHANT])))
+					return false;
+			case SKE_SUNSET_BLAST:
+				if (!(sc && (sc->data[SC_SUNSET_SUN] || sc->data[SC_SKY_ENCHANT])))
+					return false;
+			case NW_ONLY_ONE_BULLET:
+			case NW_SPIRAL_SHOOTING:
+				if (!(sd && sd->weapontype1 == W_RIFLE))
+					return false;
+			case NW_MAGAZINE_FOR_ONE:
+				if (!(sd && sd->weapontype1 == W_REVOLVER))
+					return false;
```

Comment:
Should the critical chance be half the user for NW skills? Or half critical damage?

Reply:
critical damage is halved for all skills